python/tk_multi_loader/loader_action_manager.py

```diff
@@ -268,18 +268,33 @@ def has_actions(self, publish_type):
         
     def get_actions_for_folder(self, sg_data):
         """
-        Returns a list of actions for a folder object.
+        Returns a list of actions for a folder widget.
+
+        This method is called at runtime when a folder widget with Shotgun data is selected.
+
+        :param sg_data: Standard Shotgun entity dictionary with keys type, id and name.
+        :return: List of QAction instances.
         """
-        fs = QtGui.QAction("Show in the file system", None)
-        fs.triggered[()].connect(lambda f=sg_data: self._show_in_fs(f))
-        
+
+        actions = []
+
+        # Find paths associated with the Shotgun entity.
+        paths = self._app.sgtk.paths_from_entity(sg_data["type"], sg_data["id"])
```

Comment:
It must be noted that method `paths_from_entity` returns paths stored in Shotgun as `FilesystemLocation` entities: https://github.com/shotgunsoftware/tk-core/blob/master/python/tank/api.py#L573

With my test site, the apps I used and the scripts I wrote, I could not build a proper test case in a reasonable amount of time. I will ask QA to keep an eye on the performance concern.